### PR TITLE
Update core.py, added support of NodePort k8s service type

### PR DESCRIPTION
**Pull request by black2kanoi**
added support of NodePort k8s service type

### CODE REVIEW

File: dask_kubernetes/core.py [hunk 1]
```diff
@@ -207,7 +207,22 @@ async def start(self, **kwargs):
...
+                if port.name == "dashboard":
+                    dashboard_node_port = port.node_port
+            self.external_address = "tcp://{host}:{port}".format(
+                host=hostname, port=node_port
+                )
+            self.dashboard_external_address = "http://{host}:{port}".format(
+                host=hostname, port=dashboard_node_port
+                )
+            print("External scheduler address: " + self.external_address)
+            print("External dashboard address: " + self.dashboard_external_address)
```
Comment (jacobtomlinson) [inline dask_kubernetes/core.py:225]: Please remove these print lines.

File: dask_kubernetes/core.py [hunk 2]
```diff
@@ -207,7 +207,22 @@ async def start(self, **kwargs):
             self.external_address = "tcp://{host}:{port}".format(
                 host=loadbalancer_host, port=SCHEDULER_PORT
             )
-        # FIXME Set external address when using nodeport service type
+        if self.service.spec.type == "NodePort":
+            start = time.time()
+            hostname = dask.config.get("kuberntetes.scheduler-service-nodeport-host")
```
Comment (jacobtomlinson) [inline dask_kubernetes/core.py:212]: Could you please set a default for this in the `kubernetes.yaml` configuration file.

File: dask_kubernetes/core.py (hunk 1 above)
Comment (black2kanoi) [reply dask_kubernetes/core.py]: fixed in new commit

File: dask_kubernetes/core.py (hunk 2 above)
Comment (black2kanoi) [reply dask_kubernetes/core.py]: added to defaults, and fixed typo. in new commit

**Comment by mr2dark** [comment]
Is it possible to integrate this pull request? It worked for my case with `NodePort`.
The only thing for this be consistent is that `dask.distributed` should be able to derive dashboard address (e.g. for the widget) from  `cluster.scheduler.dashboard_external_address` (if it's available) rather than from the metadata received for the scheduler service.

**Comment by jacobtomlinson** [comment]
Superseded in #304

**Pull request closed without merging**